miniapp/USERGUIDE.md

```diff
@@ -135,6 +135,20 @@ val miniAppMessageBridge = object: MiniAppMessageBridge() {
 
         callback.invoke(true)
     }
+
+    override fun requestCustomPermissions(
+        permissions: List<Pair<MiniAppCustomPermissionType, String>>,
+        callback: (grantResult: String) -> Unit
```

Comment:
Sorry, I missed this in the first PR. The callback in the public interface should be passed `List<Pair<MiniAppCustomPermissionType, MiniAppCustomPermissionResult>>` instead of a `String`. And then we can make `MiniAppCustomPermissionManager` be an internal class so the SDK can automatically handle serializing it to a JSON string for sending to the Mini App JS side.

I think this is how the iOS SDK is handling it ([see here](MiniAppCustomPermissionManager)). Is that right @ts-leojoseph-r ? iOS handles JSON serialization internally so the Host App just has to provide a specific model to the callback?

Also, the `ACCEPTED` permissions should be stored automatically by the SDK - so after the Host App provides the List of permissions to `callback.invoke`, the SDK should store them. @ts-leojoseph-r I think this is also how iOS behaves, right? The Host App doesn't need to manually call `MiniApp.setCustomPermissions` when using `MiniAppMessageBridge`?

Reply:
if HostApp doesn't need to manually call `MiniApp.setCustomPermissions`, then what is the reason to make it as public interface? 🤔

Reply:
@khairul-alam-licon `MiniApp.setCustomPermissions` and `MiniApp.getCustomPermissions` are needed so the Host App can create a Mini App Permissions screen. This screen will allow the user to manually revoke permissions to each Mini App.

Reply:
That's right @corycaywood . [MASDKCustomPermissionModel](https://github.com/rakutentech/ios-miniapp/blob/master/MiniApp/Classes/JavascriptBridge/MiniAppJavaScriptPermissionManager.swift#L140) conforms to Codable that helps the iOS SDK to serialize them into [JSON reponse](https://github.com/rakutentech/ios-miniapp/blob/master/MiniApp/Classes/JavascriptBridge/MiniAppJavaScriptPermissionManager.swift#L126).

Also, we are storing the ACCEPTED/DENIED permissions that user has responded [from the host app](https://github.com/rakutentech/ios-miniapp/blob/master/MiniApp/Classes/JavascriptBridge/MiniAppJavaScriptPermissionManager.swift#L68). Host app doesn't need to call the MiniApp.setCustomPermissions manually to store the permissions.